torch/_decomp/decompositions.py

```diff
@@ -3355,20 +3355,30 @@ def upsample_bilinear2d(
     v3 = aten._unsafe_index(input, [None, None, yp1, x])
     v4 = aten._unsafe_index(input, [None, None, yp1, xp1])
 
-    dtype = torch.float32 if not input.is_floating_point() else input.dtype
+    yscale = (y_f32 - y).clamp(0.0, 1.0).to(dtype)
+    xscale = (x_f32 - x).clamp(0.0, 1.0).to(dtype)
+
+    dtype = torch.int64 if not input.is_floating_point() else input.dtype
     if not input.is_floating_point():
         v1 = v1.to(dtype)
         v2 = v2.to(dtype)
         v3 = v3.to(dtype)
         v4 = v4.to(dtype)
 
-    yscale = (y_f32 - y).clamp(0.0, 1.0).to(dtype)
-    xscale = (x_f32 - x).clamp(0.0, 1.0).to(dtype)
+    def mul(b, a):
+        # Weights computation for uint8_t and multiplication trick
+        # Ported _compute_indices_int16_weights_aa from aten/src/ATen/native/cpu/UpSampleKernel.cpp
```

Comment:
Nit, leave a permalink to the relevant comment in the code so that it's easier to find in the future.